### PR TITLE
Player force

### PR DESCRIPTION
**Pull request by Nexela**
Adds Player and Force events, and a few other small things.
Still need to write tests for it but stuff seems to be working.

### CODE REVIEW

**Comment by Afforess** [comment]
Looks fine but at some point I'd really like tests for the queue - implementing data structures without proper testing is just asking for bugs.

File: stdlib/event/force.lua [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+--- Merge a copy of the passed data to all forces in global.forces
+-- @tparam table data a table containing variables to merge
+-- @usage local data = {a = 'abc', b = 'def'}
+-- Force.add_data_all(data)
+function Force.add_data_all(data)
+    table.each(global.forces, function(v) table.merge(v, table.deepcopy(data)) end)
+end
+
+--- Init or re-init a force or forces,
+-- @tparam[opt] string|table event table or a number containing player_index
```
Comment (Afforess) [inline stdlib/event/force.lua]: Copy and paste mistake, surely forces don't have a player_index.

File: stdlib/event/player.lua [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    local obj = {
+        index = player_index,
+        name = game.players[player_index].name,
+    }
+    return obj
+end
+
+--Print any messages in the queue.
+local function check_message_queue(player_index)
+    if global._mess_queue then
```
Comment (Afforess) [inline stdlib/event/player.lua]: This looks like partially implemented functionality ( as in, I see no way to write to `global._mess_queue` with the stdlib). I assume you are going to build on this and document it?

**Comment by Nexela** [comment]
Tests for Queue, Area, and Position are on my todo list.
Area and Position have been tweaked to allow them to be used as objects.
Area and Position are not pushed yet because I still need more core testing

local position1 = Position.to_table(entity.position)
local position2 = Positing.to_table(entity2.position)
[x] game.print(tostring(position1)) -- prints {x = X, y = Y}
[x] position1 == position2 -- false (unless it is true :))
[x] position1 + position2  -- modifies position1 by adding position2 to it
[x] position1:add(position2) -- same as above
[x] local position3 = position1:add(position2) -- same as above with position3 being a reference to position1
[x] position1:translate(0, 1)  --return position1 moved 1 tile north

Some things still left on my todo  .new() to remove the refrences and create new tables

File: stdlib/event/force.lua (hunk 1 above)
Comment (Nexela) [inline stdlib/event/force.lua]: Must have been some alternate universe :)

File: stdlib/event/player.lua (hunk 2 above)
Comment (Nexela) [inline stdlib/event/player.lua]: Correct, When it is complete it will be a buffer between init with 0 players and first created player

**Comment by Choumiko** [comment]
> [x] position1 + position2 -- modifies position1 by adding position2 to it
[x] position1:add(position2) -- same as above
[x] local position3 = position1:add(position2) -- same as above with position3 being a reference to position1

Not sure i like the idea of an addition changing one of the operands, if i understand correctly each of these operations would mutate position1 ? To me addition/subtraction are pure mathematical/functional and therefore shouldn't mess with the operands?
Also (at least for the operator overwriting) it would make addition non commutative, since
`position1 + position2 -- {1,1} + {2,2}`
`position1 + position3 -- {3,3} + {0,1} = {3,4}`
would lead to a different result than
`position2 + position1 -- {2,2} + {1,1}`
`position1 + position3 -- {1,1} + {0,1} = {1,2}`

Hope that makes sense, but as long as Position.add(pos1, pos2) still work as it does now i'm fine with it. Just recently started changing FARL to use Position, so i don't wanna touch it again too soon ;)

**Comment by Nexela** [comment]
Conversation moved to #97

**Comment by Afforess** [comment]
@Nexela Are you done with your changes here?

**Comment by Nexela** [comment]
Not quite, Need to figure out the error in player and write the tests but am on vacation with the kids for another day.

**Comment by Nexela** [comment]
I am going to clean up this PR this week and probably remove the player/force until such a time that I can write decent tests for it.

**Comment by Afforess** [comment]
I am strongly tempted to just give @Nexela push rights to the repository.

**Pull request merged**